library/src/main/java/com/getstream/sdk/chat/adapter/BaseMessageListItemViewHolder.kt

```diff
@@ -0,0 +1,43 @@
+package com.getstream.sdk.chat.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.getstream.sdk.chat.view.MessageListView.BubbleHelper
+import com.getstream.sdk.chat.view.MessageListViewStyle
+import io.getstream.chat.android.client.models.Channel
+
+abstract class BaseMessageListItemViewHolder<T : MessageListItem>(
+    resId: Int,
+    parent: ViewGroup
+) : RecyclerView.ViewHolder(
+    LayoutInflater.from(parent.context).inflate(resId, parent, false)
+) {
+
+    /**
+     * Workaround to allow a downcast of the MessageListItem to T
+     */
+    fun bindListItem(
```

Comment:
This method seems to have lot of arguments and I think not all of them are needed to bind an item to its view.
`context` is not needed because we have an already inflated view where we can get the context.
A `ViewHolder` shouldn't need the factory that already creates itself
The rest of arguments, if they are needed to the render process, should be into our `MessageListItem`

Reply:
This is covered by a separate task, CAS-231 🙂